### DISCUSSION TITLE
failure with parallel computation with WENO

### DISCUSSION THREAD

**Discussion by Woo-ZL** [WENO-OF/WENOEXT#41] [Q&A] [answered]
Hi all,

I encounter a problem with using the WENO scheme. I ran the tutorials well with the scheme in parallelization. However, when I ran a case in parallelization with a mesh generated by myself, it always failed with the following error. But it works when calculating with a single processor.

![error](https://user-images.githubusercontent.com/105489606/168258099-1ef27703-c1bf-4460-b6f8-9f8838e1afee.jpg)

Kind regards,

**Comment by JanGaertner** [comment, marked as answer]
Dear Zhenlong,

did you use the master branch? I have made some improvements on the development branch to fix these kind of issues especially when you use collated file I/O option in OpenFOAM.

The version on the master branch requires that the case is decomposed with the uncollated format and that the processor directories can be read. The new version on the development branch circumvents this by actually sending the mesh information over MPI to the required processors. This has two advantages,

 1. On large cases reading from the hard drive takes a long time and is now no longer needed
 2. Collated and uncollated file I/O can be used

In summary, please try out the development branch :)

**Comment by Woo-ZL** [comment]
Dear Jan,

Thanks a lot for your prompt and detailed explanation. Yes, I am using the old version. I will change to the development version to see if it helps. Thank you again.

Kind regards,
Zhenlong Wu
